packages/core/src/modules/dids/__tests__/peer-did.test.ts

```diff
@@ -134,17 +139,20 @@ describe('peer dids', () => {
       expect(didPeer.didDocument.toJSON()).toMatchObject(didPeer1zQmY)
     }
 
-    // If the method is a genesis doc (did:peer:1) we should store the document
-    // Otherwise this is not needed as we can generate it form the did itself
-    if (didPeer.numAlgo === PeerDidNumAlgo.GenesisDoc) {
-      const didDocumentRecord = new DidDocumentRecord({
-        id: didPeer.did,
-        role: DidDocumentRole.Received,
-        didDocument: didPeer.didDocument,
-      })
+    const didDocumentRecord = new DidRecord({
+      id: didPeer.did,
+      role: DidDocumentRole.Received,
+      // If the method is a genesis doc (did:peer:1) we should store the document
+      // Otherwise we only need to store the did itself (as the did can be generated)
+      didDocument: didPeer.numAlgo === PeerDidNumAlgo.GenesisDoc ? didPeer.didDocument : undefined,
+      tags: {
+        // We need to save the recipientKeys, so we can find the associated did
+        // of a key when we receive a message from another connection.
+        recipientKeys: didPeer.didDocument.recipientKeys,
+      },
+    })
```

Comment:
This means we now always need to create a did record (that's why I changed it from did document record to did record), but only if using `numAlgo === PeerDidNumAlgo.GenesisDoc` we store the actual did document